telemetry/csharp/ToolkitTelemetryGenerator/DefinitionsBuilder.cs

```diff
@@ -0,0 +1,757 @@
+﻿using System;
+using System.CodeDom;
+using System.CodeDom.Compiler;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using ToolkitTelemetryGenerator.Models;
+using ToolkitTelemetryGenerator.Utils;
+
+namespace ToolkitTelemetryGenerator
+{
+    /// <summary>
+    /// Generates code that allows programs like the Toolkit to instantiate and publish Telemetry Events
+    /// </summary>
+    public class DefinitionsBuilder
```

Comment:
Adding @kiiadi 's top level comment here so we have threading:

> We should have tests that for a sample definition input we get the expected code generated - it also helps with code reviews because you can see the 'expected file' (so no need to paste a copy-paste in the description).

Reply:
This is a good call, I will add this. People tend to ask to see generated output regardless, which is why I put it in the PR description.

Reply:
in that case you can just point em at the tests :P

Reply:
what would be great in the PR description is an example of how to use the generated stuffs.

Reply:
Done